### DISCUSSION TITLE
Revoke OAuth2 Token

### DISCUSSION THREAD

**Discussion by sevontheweb** [ory/hydra#2776] [Q&A] [answered]
The following documentation pertains to the functionality of revoking an OAuth2 Token.
https://www.ory.sh/hydra/docs/reference/api/#operation/revokeOAuth2Token

Revoking tokes is also on the OAuth 2.0 Specification: https://oauth.net/2/token-revocation/

Nevertheless, it seems that Ory Hydra either does not implement the endpoint (?) or the documentation is not correct.
A call to this endpoint based on the specification results in an error.
I did not open a bug request because during my investigation of the issue I read multiple times that there is no use case for token revocation (since tokens are not sessions, etc., etc.).

So the question here is double-folded:
a) should a client upon logout revoke the associated token
b) if yes, either documentation or implementation in Ory Hydra is broken. How/Where to properly report it?

Many thanks.

**Comment by vinckr** [comment, marked as answer]
a) should a client upon logout revoke the associated token

No, as you rightly pointed out the OAuth2 token is not a session token. 

> You will use OpenID Connect to initiate and complete the login, but your application needs to store and track and invalidate the session.
>Ory Hydra also maintains a session cookie.

You obviously read the "[Before OAuth2](https://www.ory.sh/hydra/docs/next/concepts/before-oauth2)" and the rest of the documentation, so I am wondering where we could make it a bit more clearer. 
In the Before OAuth2 document is an example with GitHub and CircleCI that explains why you don't revoke the token on logout. Key sentence:

> It would be crazy if some app could just log you out of GitHub, or if signing out of Google would invalidate your sessions on all the websites where you used "Sign in with Google".

There is a use case for token revocation, 

---

>  A call to this endpoint based on the specification results in an error.

The endpoint works as outlined in the specification otherwise the tests would not pass, so this could be a misconfigured request or bug.

**Reply by vinckr** [reply]
Hey @sevontheweb ,
Just checking in if your question was answered or if there was still something unclear.
I will mark this as answered for now, but feel free to come back to the discussion any time.

Thanks a ton! 🐝
